app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CategoriesController < ApplicationController
+
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def check_for_duplicates(category)
+    if Category.exists?(title: category.title)
+      flash[:error] = "#{category.title} already exists."
+    end
+  end
+
+  def create
+    @category = Category.new(category_params)
+    check_for_duplicates(@category)
+    if @category.save
+      flash[:success] = "#{@category.title} added!"
+      redirect_to category_path(@category)
+    else
+      redirect_to new_category_path
+    end
+  end
+
+  def show
+    @category = Category.find(params[:id])
+    @jobs = Job.where(category_id: @category.id)
```

Comment:
we can and should leverage our associations here instead by calling -

 `@jobs = @category.jobs`